### ISSUE TITLE
[Build Failure]: Installation fails with --with-native-comp: The installed libgccjit failed to compile and run

### ISSUE THREAD

**Issue by elondres-mim** [d12frosted/homebrew-emacs-plus#684]
### Version

emacs-plus@29

### Make sure to follow these steps before submitting the issue

- [X] run `brew update` and try to reproduce the issue again
- [X] run `brew doctor`, fix all issues and try to reproduce your issue again
- [X] try to reproduce your issue without any custom options to the formula

### What happened?

Hi, I originally posted this in the other issue but it was already closed, so I'm making a new one for my new problem.

<pre>
$ brew install emacs-plus --with-native-comp
</pre>

<details>
  <summary>Command output</summary>

  <pre>
# ... various configure checks ...
checking for gcc_jit_context_acquire in -lgccjit... yes
checking for libgccjit.h... yes
configure: error: The installed libgccjit failed to compile and run a test program using
the libgccjit library; see config.log for the details of the failure.
The test program can be found here:
<https://gcc.gnu.org/onlinedocs/jit/intro/tutorial01.html>.
You can try compiling it yourself to investigate the issues.
Please report the issue to your distribution if libgccjit was installed
through that.
You can find the instructions on how to compile and install libgccjit from
source on this site:
<https://gcc.gnu.org/wiki/JIT>.

If reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core):
  https://github.com/d12frosted/homebrew-emacs-plus/issues

  </pre>
</details>

<details>
<summary>Error lines from config.log</summary>
<pre>
5221   │ configure:20509: $? = 0
5222   │ configure:20509: ./conftest
5223   │ dyld[21515]: Library not loaded: @rpath/libgccjit.0.dylib
5224   │   Referenced from: <B83F28CD-9C18-3332-BA5E-B8042F6963A9> /private/tmp/emacs-plusA29-20240531-19053-bt1kwq/emacs-29.3/confte
       │ st
5225   │   Reason: no LC_RPATH's found
5226   │ ./configure: line 3448: 21515 Abort trap: 6           ./conftest$ac_exeext
5227   │ configure:20509: $? = 134
5228   │ configure: program exited with status 134
5</pre></details>

### brew doctor output

Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed casks are deprecated or disabled.
You should find replacements for the following casks:
  wkhtmltopdf

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  cowsay
  openssl@1.1
  packer
  terraform
  youtube-dl

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libwkhtmltox.0.12.6.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected header files:
  /usr/local/include/wkhtmltox/image.h
  /usr/local/include/wkhtmltox/pdf.h

### brew config output

HOMEBREW_VERSION: 4.3.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: e130e47f23b8b806096f9ec4f2c193213b8ec908
Last commit: 2 days ago
Core tap JSON: 04 Jun 13:05 UTC
Core cask tap JSON: 04 Jun 13:05 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: /Applications/MacPorts/EmacsMac.app/Contents/MacOS/bin/emacsclient
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 3.3.2 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.2/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.45.2 => /usr/local/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-x86_64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.4

**Comment by d12frosted** [comment]
To be honest, I have no ideas. I saw that you tried reinstalling `libgccjit` and it didn't help, so let's try one of the following things:

1. Try compiling the [Hello World](https://gcc.gnu.org/onlinedocs/gcc-14.1.0/jit/intro/tutorial01.html) example. If it works, then at least we know that `libgccjit` is properly installed and your env is properly configured.
2. Try removing `:env std` line from the formula (see below) by using `brew edit emacs-plus`. And try installing again.

https://github.com/d12frosted/homebrew-emacs-plus/blob/61a1e971b228f416a95374cb8e5d95631fff3a42/Formula/emacs-plus%4029.rb#L9-L11

**Comment by elondres-mim** [comment]
Thank you for the quick response!

I tried compiling the hello world example and got an error:
```
ld: library 'gccjit' not found
```

Googling this error the first result was #338 which seems to have been related to Apple Silicon users a few years ago when libgccjit on homebrew didn't support it. I'm on an Intel mac, though.

I tried `brew install libgccjit --build-from-source`, which took a very long time, but I got the same error on the hello world example.

So maybe this is a libgccjit problem, and not an emacs-plus problem?

**Comment by d12frosted** [comment]
Could be something either with libgccjit or with your env. Please try the second option as well (i.e. modify the formula by removing `env :std`. If it compiles without these lines - something is amiss in your env.

**Comment by d12frosted** [comment]
No, it does not :) I still don't know why it doesn't work for you, but at least we confirmed two things:

1. You can't use installed libgccjit
2. Even super env (default in brew) doesn't help

So it does sound like the issue is with libgccjit formula and/or your env. 

I know it's not helpful 😢 All I can say is that native compilation is the worst thing happened to Emacs+ maintenance-wise as all this libgccjit business is highly unreliable. Hence I stopped using it myself 🤷

**Comment by d12frosted** [comment]
Ok, it's easy to complain, so instead decided to give another try to fixing compilation.

Can you please try two more times?

1. Apply the changes from https://github.com/d12frosted/homebrew-emacs-plus/pull/686. Theoretically it should work with emacs-plus@29, but I would prefer you trying out the emacs-plus@30
2. Same as (1) but also remove `env :std`.

🙏

**Comment by d12frosted** [comment]
FYI I've merged the linked PR.

**Comment by elondres-mim** [comment]
@d12frosted That worked! I just `brew install emacs-plus@30 --with-native-comp` and everything succeeded!

Thank you so much!
